server/actions_specs.go

```diff
@@ -198,6 +203,46 @@ func (s *RegistryServer) getApiSpecRevision(ctx context.Context, name names.Spec
 	return message, nil
 }
 
+// GUnzippedBytes uncompresses a slice of bytes.
+func GUnzippedBytes(input []byte) ([]byte, error) {
+	buf := bytes.NewBuffer(input)
+	zr, err := gzip.NewReader(buf)
+	if err != nil {
+		return nil, err
+	}
+	return ioutil.ReadAll(zr)
+}
+
+// GetApiSpecContents handles the corresponding API request.
+func (s *RegistryServer) GetApiSpecContents(ctx context.Context, req *rpc.GetApiSpecContentsRequest) (*httpbody.HttpBody, error) {
+	var spec *rpc.ApiSpec
+	var err error
+	if name, err := names.ParseSpec(req.GetName()); err == nil {
+		spec, err = s.getApiSpec(ctx, name, rpc.View_FULL)
+	} else if name, err := names.ParseSpecRevision(req.GetName()); err == nil {
+		spec, err = s.getApiSpecRevision(ctx, name, rpc.View_FULL)
+	}
+	if err != nil {
+		return nil, err
+	}
+	if spec != nil {
+		contents := spec.Contents
+		contentType := spec.MimeType
```

Comment:
Nit: These seem to slightly obscure the data we're operating on in the next few lines. Could we just reference MimeType and Contents directly?

Reply:
Revised in the latest version.